bucket_map/src/bucket.rs

```diff
@@ -366,22 +367,30 @@ impl<T: Clone + Copy> Bucket<T> {
         //debug!(            "INDEX_IX: {:?} uid:{} loc: {} cap:{}",            key,            uid,            location,            index.capacity()        );
     }
 
+    /// if result is a grow error, then grow the appropriate piece and return true
+    pub fn grow_from_error(&mut self, result: Result<(), BucketMapError>) -> bool {
```

Comment:
I get the intent here, but it feels like this needs some massaging. Because even though the name is "grow from error", it doesn't _always_ grow, and the parameter _isn't_ an error, it's a result (which may not be an error).

Maybe this fn can be 
```fn grow(&mut self, err: BucketMapError) { ... }```
so that it'll always grow. Then in `insert()`, it could be (inside the `loop`):

```
match self.try_write(key, new, refct) {
    Ok(()) => return,
    Err(err) => self.grow(err),
}
```

Reply:
> I'm not sure I understand the need for adding the param to `insert()` and `try_insert()`, but not to `update()`. The bucket_map API now feels inconsistent both with itself, and with HashMap (not saying it needs to be, but an observation).
> 
> Maybe there's a new fn in bucket_map to get the `Bucket` from the bucket ix, and then the caller has a `Bucket` object to call `insert` on. Would that work?
> 
> (If not, maybe new functions like `insert_in_bucket()` and `try_insert_in_bucket()` could work?)

yes, some inconsistencies. update is not used by any current callers anymore. Insert won't be used either soon (replaced by try_insert). Ultimately, we can add a self-ref struct which allows us a better apis behind RwLock

Reply:
renamed grow